src/aatoolbox/utils/raster.py

```diff
@@ -0,0 +1,447 @@
+"""
+Utilities to manipulate and analyze raster data.
+
+This module extends xarray and rioxarray to provide
+additional functionality for raster processing and
+post-processing. The extension is based on the
+guidance for how to extend xarray:
+
+http://xarray.pydata.org/en/stable/internals/extending-xarray.html
+
+However, since rioxarray already extends xarray, this
+modules extensions inherit from the RasterArray and
+RasterDataset extensions respectively. This ensures
+cleaner code in the module as ``rio`` methods are
+available immediately, but also means a couple of
+design decisions are followed.
+
+The xarray.DataArray and xarray.DataSet
+extensions here inherit from rioxarray base classes.
+Thus, methods that are identical for both objects
+are defined in a mixin class ``AatRasterMixin`` which
+can be inherited by the two respective extensions.
+"""
+
+import logging
+from typing import Any, Callable, Dict, List, Union
+
+import geopandas as gpd
+import numpy as np
+import pandas
+import pandas as pd
+import rioxarray
+import xarray
+import xarray as xr
+from rioxarray.raster_array import RasterArray
+from rioxarray.rioxarray import _get_data_var_message
+
+logger = logging.getLogger(__name__)
+
+
+class AatRasterMixin:
+    """AA toolbox mixin base class."""
+
+    # setting attributes to avoid mypy error, from SO
+    # https://stackoverflow.com/questions/53120262/mypy-how-to-
+    # ignore-missing-attribute-errors-in-mixins/53228204#53228204
+    _get_obj: Callable
```

Comment:
That's a good point. I have no idea actually. Seems a bit strange/nonsensical for one to be caught but another not to be. Could it be because if `_obj` doesn't exist it will default to `None` while there's no default if a function/Callable doesn't exist?

Reply:
At some point, mypy called out `_obj` as well, so also added there. Strange.